### ISSUE TITLE
Cron job does not work with Maps app enabled

### ISSUE THREAD

**Issue by R7JANV1** [nextcloud/maps#691]
https://github.com/nextcloud/all-in-one/issues/318

For all with the same problem:
If the Maps app is deactivated and reactivated, the same problem.

_Originally posted by @R7JANV1 in https://github.com/nextcloud/all-in-one/issues/318#issuecomment-1060447734_

**Comment by call-me-matt** [comment]
Same problem here:

```
DivisionByZeroError: Division by zero

    /var/www/html/custom_apps/maps/lib/Service/PhotofilesService.php - line 264:

    OCA\Maps\Service\PhotofilesService->getExif(OC\Files\Node\File {})

    /var/www/html/custom_apps/maps/lib/BackgroundJob/AddPhotoJob.php - line 56:

    OCA\Maps\Service\PhotofilesService->addPhotoNow(OC\Files\Node\File {}, "matthias")

    /var/www/html/lib/public/BackgroundJob/Job.php - line 79:

    OCA\Maps\BackgroundJob\AddPhotoJob->run({ photoId: 1 ... "})

    /var/www/html/lib/public/BackgroundJob/QueuedJob.php - line 47:

    OCP\BackgroundJob\Job->execute(OC\BackgroundJob\JobList {}, OC\Log {})

    /var/www/html/cron.php - line 128:

    OCP\BackgroundJob\QueuedJob->execute(OC\BackgroundJob\JobList {}, OC\Log {})
```

**Comment by blackie75** [comment]
I am not an IT specialist or a programmer but I have found a solution that may be acceptable for some people. 


1. Move all photos outside of Nextcloud
2. Select CRON in Nextcloud and run the command in the console
sudo -u www-data php -f /var/www/html/nextcloud/cron.php 

If no error message appears, you can proceed to the next step

3. Using the EXIFTOOL tool generate a csv file (I used the basic parameters of the EXIFTOOL tool such as path, directory, latitude, longitude and date of the photo creation. This tool will allow you to extract files with correct EXIF ​​data (because, as you know, CRON stops when it comes across files with incorrect EXIF ​​data),
4. Open the csv file in Excel, then save it in the default xls format and create a pivot table that will make your work easier and save some time,
5. When selecting individual photo catalogs in the pivot table (I assume that you catalog photos), compare the contents of the pivot table with the photo catalogs transferred from Nextcloud.
6. Send photos to Nextcloud according to the list in the pivot table
7. Install and start the MAPS application. Selected photos with correct EXIF ​​data will appear on the maps and CRON will not stop due to the launch of the MAPS application

Summary: In my case, it took an hour to browse catalogs containing 3000 photos. The photos filtered and uploaded in this way are a guarantee of the correct operation of CRON with the MAPS application running
You can probably do it even faster by using the available Excel functions, comparing csv files with the full contents of the directory with the data obtained with EXIFTOOL - I have not tried yet. I will also add that I was browsing the forum before and tried virtually all the tricks (including editing files like PhotofilesService.php) and got no solution to the problem. It was only after filtering the files and getting rid of those with wrong EXIF ​​data that the CRON problem disappeared.There is still a solution to the problem with files with invalid EXIF ​​data. Fortunately, in my case, the percentage of such files was negligible.

My settings: [Nextcloud Hub II] (23.0.3), PHP 8.0.17

**Comment by tacruc** [comment]
Can you check if  #586 resolves your issue?

**Comment by R7JANV1** [comment]
Could work.  When can we expect the next update?

**Comment by blackie75** [comment]
@tacruc As I wrote earlier, I am neither a programmer nor an IT specialist. Yes, I also tried to fix the problem using the link provided. Unfortunately, in my case it didn't solve the problem - maybe because I'm using php 8. If I understood correctly, this solution requires downgrading php to 7.3. It is also possible that I was not focused enough editing the files.I am glad that there are passionate people who share their knowledge, but having no experience, it is difficult to use this knowledge. That is why I found an intermediate solution that satisfies me so far.
